pkg/api/auth.go

```diff
@@ -1,277 +1,17 @@
-	if redirectURI != "" {
-		loginPage = redirectURI
-	}
-
-	if strings.HasPrefix(ctx.Request.Host, "localhost") {
-		loginPage = "http://localhost:8080" + loginPage
-	}
-
-	deleteCookie(ctx, RedirectURICookie, host)
-	code := ctx.Request.URL.Query().Get("code")
-	if len(code) == 0 {
-		return loginPage + "?error=unauthenticated", errors.New("unauthenticated")
-	}
-
-	oauthCookie, err := ctx.Cookie(OAuthStateCookie)
-	if err != nil {
-		c.log.Infof("Missing oauth state cookie: %v", err)
-		return loginPage + "?error=invalid-state", errors.New("invalid state")
-	}
-
-	deleteCookie(ctx, OAuthStateCookie, host)
-
-	state := ctx.Request.URL.Query().Get("state")
-	if state != oauthCookie {
-		c.log.Info("Incoming state does not match local state")
-		return loginPage + "?error=invalid-state", errors.New("invalid state")
-	}
-
-	tokens, err := c.azureClient.Exchange(ctx.Request.Context(), code)
-	if err != nil {
-		if !errors.Is(err, context.Canceled) {
-			c.log.Errorf("Exchanging authorization code for tokens: %v", err)
-		}
-		return loginPage + "?error=invalid-state", errors.New("forbidden")
-	}
-
-	rawIDToken, ok := tokens.Extra("id_token").(string)
-	if !ok {
-		c.log.Info("Missing id_token")
-		return loginPage + "?error=unauthenticated", errors.New("unauthenticated")
-	}
-
-	// Parse and verify ID Token payload.
-	_, err = c.azureClient.Verify(ctx.Request.Context(), rawIDToken)
-	if err != nil {
-		c.log.Info("Invalid id_token")
-		return loginPage + "?error=unauthenticated", errors.New("unauthenticated")
-	}
-
-	session := &auth.Session{
-		Token:       generateSecureToken(tokenLength),
-		Expires:     time.Now().Add(sessionLength),
-		AccessToken: tokens.AccessToken,
-	}
-
-	b, err := base64.RawStdEncoding.DecodeString(strings.Split(tokens.AccessToken, ".")[1])
-	if err != nil {
-		c.log.WithError(err).Error("unable decode access token")
-		return loginPage + "?error=unauthenticated", errors.New("unauthenticated")
-	}
-
-	if err := json.Unmarshal(b, session); err != nil {
-		c.log.WithError(err).Error("unable unmarshalling token")
-		return loginPage + "?error=unauthenticated", errors.New("unauthenticated")
-	}
-
-	session.IsAdmin = c.isUserInAdminGroup(session.AccessToken)
-
-	if err := c.repo.SessionCreate(ctx, session); err != nil {
-		c.log.WithError(err).Error("unable to create session")
-		return loginPage + "?error=internal-server-error", errors.New("unable to create session")
-	}
-
-	ctx.SetCookie(
-		sessionCookie,
-		session.Token,
-		86400,
-		"/",
-		host,
-		true,
-		true,
-	)
-
-	return loginPage, nil
-}
-
-func (c *client) logout(ctx *gin.Context) (string, error) {
-	host, _, err := net.SplitHostPort(ctx.Request.Host)
-	if err != nil {
-		host = ctx.Request.Host
-	}
-
-	deleteCookie(ctx, sessionCookie, host)
-
-	var loginPage string
-	if strings.HasPrefix(ctx.Request.Host, "localhost") {
-		loginPage = "http://localhost:8080/"
-	} else {
-		loginPage = "/"
-	}
-
-	err = c.repo.SessionDelete(ctx, sessionCookie)
```

Comment:
Since `sessionCookie` is always `knorten_session`, we are never actually removing a session from the database as far as I can tell?

Reply:
What we should be doing here is retrieving the cookie value, which is the session.Token, and then we should be sending the session.Token to the database for deletion